### DISCUSSION TITLE
iOS Out of memory (OOM) error handling in Crashlytics

### DISCUSSION THREAD

**Discussion by hovox** [firebase/firebase-ios-sdk#8130] [Q&A]
Hi all,

There are some crashes on iOS which are not emitting any signals. The most common cases are OOMs and app hangs. Those crashes are not handled by Crashlytics, so Crashlytics users(we :) ) currently do not have full picture of app crashes on iOS. There are some ways of handling this kind of crashes, for example [here](https://github.com/bugsnag/bugsnag-cocoa/blob/b83847c74669f64a9afc0e7e7ade595b620c09e8/Bugsnag/Client/BugsnagClient.m#L476) we can see how bugsnag handles those cases. Currently we workaround this by sending analytics in that cases, but analytics is not as convenient as Crashlytics for tracking crashes. While we will not have stack traces, there are a lot of other info by which we can track this kind of issues(logs, custom keys, standard keys).

I wonder is there any internal plans in the Crashlytics roadmap to add this kind of thing, if not what you are thinking about adding this?

Edit: Hi there, I would like to get opinions/suggestions/thoughts from the firebase team about this topic. With Xcode 13 apple has added `Terminations` tab, which I believe will should what I have described here.

**Comment by gonzo-oin** [comment]
Hello!

I'm also very interested about this feature. It's very disturbing to have users reporting app crashes and see nothing in Firebase.

> Currently we workaround this by sending analytics in that cases, but analytics is not as convenient as Crashlytics for tracking crashes. 
@hovox I'm curious about how you do that ? How you differentiate users that kill the app and OOM and app hangs ?

**Reply by hovox** [reply]
You can checkout code in Bugsnag and other crash reporting tools, so basically the idea is that if app was in foreground while terminated, then it is a oom or other crash. When user or system kills the app, system will call `appWillTerminate...` callback method, so all other cases are considered crash terminations.

**Reply by gonzo-oin** [reply]
Thank you @hovox for the advice. It's seems pretty simple, it's weird Firebase don't handle that kind of crash.
